server/test/admin.spec.js

```diff
@@ -0,0 +1,121 @@
+import chaiHttp from 'chai-http';
+import chai, { expect } from 'chai';
+import dotenv from 'dotenv';
+import app from '../app';
+
+dotenv.config();
+chai.use(chaiHttp);
+
+let superAdminToken;
+let userToken;
+
+
+describe('Superadmin Endpoints', () => {
+  before((done) => {
+    const superAdminData = {
+      name: process.env.SUPER_ADMIN_USERNAME,
+      password: process.env.SUPER_ADMIN_PASSWORD
+    };
+    chai
+      .request(app)
+      .post('/api/users/login')
+      .send(superAdminData)
+      .end((err, res) => {
+        superAdminToken = res.body.user.token;
+        const userData = {
+          name: 'ayo',
+          password: 'admin123456'
+        };
+        chai
+          .request(app)
+          .post('/api/users/login')
+          .send(userData)
+          .end((error, response) => {
+            userToken = response.body.user.token;
+            done();
+          });
+      });
+  });
+  it('should allow a superadmin to give admin permissions', (done) => {
+    chai
+      .request(app)
+      .patch('/api/admin/ayo/upgrade')
```

Comment:
The other comment below  also applies here for ayo

Reply:
'ayo' is the username of one of the users seeded into the DB without admin privileges so instead of creating a new user to test the feature I used the existing user